### PR TITLE
Zipkin to operate in node > v8

### PR DESCRIPTION
**Pull request by pbadenski**
zipkin-js source in version 1.X will be written and published for node > v8.

This was previously discussed with @adriancole @SimenB @eirslett (sorry if I missed someone).

Unblocks https://github.com/openzipkin/zipkin-js/pull/135, https://github.com/openzipkin/zipkin-js/pull/208 and https://github.com/openzipkin/zipkin-js/pull/250

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,2 @@
 {
```
Comment (SimenB) [inline .babelrc]: remove the whole file?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 language: node_js
 node_js:
   - 8
```
Comment (SimenB) [inline .travis.yml]: 9 is released, should probably add it

File: package.json [hunk 3]
```diff
@@ -17,7 +17,6 @@
   "repository": "https://github.com/openzipkin/zipkin-js",
   "devDependencies": {
     "babel-cli": "^6.23.0",
```
Comment (SimenB) [inline package.json]: are the other babel deps needed?

**Comment by SimenB** [comment]
What about build scripts and `main`s in `package.json`s?

**Comment by eirslett** [comment]
- [ ] npmignore must be updated
- [ ] gitignore must be updated
- [ ] all build scripts in npm must be removed
- [ ] remove babel hooks from mocha

File: .babelrc (hunk 1 above)
Comment (pbadenski) [reply .babelrc]: Will do.

File: .travis.yml (hunk 2 above)
Comment (pbadenski) [reply .travis.yml]: 👍 will do

**Comment by pbadenski** [comment]
I'm wondering if it wouldn't be better to leave babel with empty presets? I'm suspecting we would like need it back soon for browser support? Unless it would be the user's responsibility to transpile zipkin for their needs in browser context (which I reckon isn't very common).

**Comment by vigneshnrfs** [comment]
Any eta on this being merged into master?

**Comment by adriancole** [comment]
There are open questions in the comments here. We need to answer these
before proceeding

**Comment by adriancole** [comment]
pitching the idea to drop node <8

reason is that we are fast accumulating debt where libraries and such are dependent on it anyway. Being neither here nor there hasn't magically resolved meanwhile. node is the "higher priority" library type and we could consider throwing some things under the bus if needed, if we agree anyway.

Should we just do it? If so, what needs to be dropped, if anything. I'm concerned lack of any movement will cripple folks long term, moreso than being conservative would enable folks.

cc @openzipkin/core

**Comment by jcchavezs** [comment]
Sounds reasonable to me. However I would leave this as open to see if
people has any concern. Would be also nice to ping people here that might
be using zipkin-js to see if there is anything else that could be dropped
or if a concern can be raised.

Ping @epallerols @fbaiodias
-- 

*José Carlos*

**Comment by jcchavezs** [comment]
@pbadenski  can we rebase this?

**Comment by jcchavezs** [comment]
Ping @timneutkens

**Comment by timneutkens** [comment]
@jcchavezs I'm not sure why you're @mentioning me 🤔 I have never seen this project before.

**Comment by jcchavezs** [comment]
I rebased this PR. I would like that you raise the concerns.

To be honest, before we merge this I would like that we look at what could be broken for browser and more over, make sure we have the right tests for anything could be broken.

Pending questions/actions are:

1. Checklist form @eirslett:

- [ ] npmignore must be updated
- [ ] gitignore must be updated
- [ ] all build scripts in npm must be removed
- [ ] remove babel hooks from mocha

2. Questions we should address are:
> I'm wondering if it wouldn't be better to leave babel with empty presets? I'm suspecting we would like need it back soon for browser support? Unless it would be the user's responsibility to transpile zipkin for their needs in browser context (which I reckon isn't very common).

Ping @pbadenski @mackermans @SimenB @eirslett @vigneshnrfs

**Comment by timneutkens** [comment]
> @openzipkin openzipkin (Open Zipkin) deleted a comment from timneutkens 7 days ago

I'd very much appreciate an explanation as to why I got pinged here instead of removing my comment. As I'm still getting notifications.

**Comment by jcchavezs** [comment]
Sorry @timneutkens I pinged you by mistake and tried to contact you by gitter to apologize (I think you are not in gitter). I removed your comment because I though that would stop the noise for you but apparently not.

In the other hand you seem to be an amazing developer in javascript and react native so if you wanna participate of this nice community you are moooore than welcome, we have plenty of questions and I believe you know most of the answers.

**Comment by timneutkens** [comment]
No worries 🙏 Thanks for the explanation 👍 I'm indeed not using gitter.

**Comment by jcchavezs** [comment]
@pbadenski any chance we can continue with this?

**Comment by jcchavezs** [comment]
Ping @ghermeto

**Comment by jcchavezs** [comment]
@pbadenski since [we dropped support for node 6](https://github.com/openzipkin/zipkin-js/pull/344), is this PR still valid?

**Comment by jcchavezs** [comment]
I think most of the concerns of this PR have been addressed in the latest changes. I am closing it but feel free to reopen if needed.

**Pull request closed without merging**